rclpy/test/test_select_rmw_implementation.py

```diff
@@ -0,0 +1,137 @@
+# limitations under the License.
+
+import ctypes
+import multiprocessing
+import os
+import traceback
+
+# TODO(wjwwood): remove this once we get all the identifiers consistent
+identifiers_map = {
+    'connext_static': 'rmw_connext_cpp',
+    'opensplice_static': 'rmw_opensplice_cpp'
+}
+
+
+def run_catch_report_raise(func, *args, **kwargs):
+    try:
+        return func(*args, **kwargs)
+    except:
+        print("exception in {0}():".format(func.__name__))
+        traceback.print_exc()
+        raise
+
+
+def func_import_each_available_rmw_implementation(rmw_implementation):
+    import rclpy
+    from rclpy.impl.rmw_implementation_tools import get_rmw_implementations
+    from rclpy.impl.rmw_implementation_tools import select_rmw_implementation
+
+    rmw_implementations = get_rmw_implementations()
+    assert(rmw_implementation in rmw_implementations)
+
+    select_rmw_implementation(rmw_implementation)
+    try:
+        rclpy.init([])
+    except ImportError as exc:
+        if "No module named 'rclpy._rclpy__" in '{0}'.format(exc):
+            # Not all rmw implementations generate a Python implementation.
+            # TODO(wjwwood): prune get_rmw_implementations by
+            #   implementations that have generated a Python implementation
+            #   (probably use a separate ament index entry).
+            print("Failed to import rmw implementation '{0}', ignoring."
+                  .format(rmw_implementation))
+            return ctypes.c_bool(True)
+        raise
+
+    set_rmw = rclpy.get_implementation_identifier()
+
+    assert set_rmw in identifiers_map, \
+        "expected identifier '{0}' to be in the map '{1}'".format(set_rmw, identifiers_map)
+
+    set_rmw_equivalent = identifiers_map[set_rmw]
+    assert set_rmw_equivalent == rmw_implementation, \
+        "expected '{0}' but got '{1}' (aka '{2}')".format(
+            rmw_implementation, set_rmw_equivalent, set_rmw)
+    return ctypes.c_bool(True)
+
+
+def test_import_each_available_rmw_implementation():
+    from rclpy.impl.rmw_implementation_tools import get_rmw_implementations
+    for rmw_implementation in get_rmw_implementations():
+        pool = multiprocessing.Pool(1)
+        result = pool.apply(
+            func=run_catch_report_raise,
+            args=(func_import_each_available_rmw_implementation, rmw_implementation,)
+        )
+
+        assert result
+        pool.close()
+        pool.join()
+
+
+def func_select_rmw_implementation_by_environment(rmw_implementation):
+    import rclpy
+    from rclpy.impl.rmw_implementation_tools import get_rmw_implementations
+
+    os.environ['RCLPY_IMPLEMENTATION'] = rmw_implementation
+
+    assert('RCLPY_IMPLEMENTATION' in os.environ)
+
+    rmw_implementations = get_rmw_implementations()
+    assert(os.environ['RCLPY_IMPLEMENTATION'] in rmw_implementations)
+
+    try:
+        rclpy.init([])
+    except ImportError as exc:
+        if "No module named 'rclpy._rclpy__" in '{0}'.format(exc):
```

Comment:
Since it's not necessary in Python 3 I think simplifying the code would make it more readable:

    '{0}'.format(exc)

vs.

    str(exc)